src/Common/src/TypeSystem/Common/Utilities/GCPointerMap.Algorithm.cs

```diff
@@ -61,7 +61,7 @@ public static GCPointerMap FromStaticLayout(DefType type)
         /// </summary>
         public static GCPointerMap FromThreadStaticLayout(DefType type)
         {
-            GCPointerMapBuilder builder = new GCPointerMapBuilder(type.ThreadStaticFieldSize.AsInt, type.Context.Target.PointerSize);
+            GCPointerMapBuilder builder = new GCPointerMapBuilder(type.ThreadGcStaticFieldSize.AsInt, type.Context.Target.PointerSize);
```

Comment:
The `foreach` below will go over all threadstatic fields, but it should only go over fields with a GC static base.

If you take my other suggestion about adjusting what `HasGCStaticBase` says, we can add `!field.HasGCStaticBase` to the `if` block below and fix it.

It feels like this would not be a GC hole right now, but I would rather we just fix it - especially because this change is not adding unit tests (we test the CoreRT behaviors in `GCPointerMapTests.cs`).